src/app/controllers/course_controller.rb

```diff
@@ -0,0 +1,36 @@
+class CourseController < ApplicationController
+  before_action :require_teacher, only: [:edit, :update, :destroy]
+
+  def index
+    @courses = Course.all
+  end
+
+  def show
+    @course = Course.find(params[:id])
+  end
+
+  def edit
+    @course = Course.find(params[:id])
+  end
+
+  def update
+    @course = Course.find(params[:id])
+      if @course.update(course_params)
+        redirect_to @course
+      else
+        render 'edit'
+      end
+  end
+
+  def destroy
+    @course = Course.find(params[:id])
+    @course.destroy
+    redirect_to root_url
```

Comment:
I think we should redirect to the course list or to the dashboard instead of root